website/docs/cdktf/python/r/lb_target_group.html.markdown

```diff
@@ -146,7 +146,8 @@ This resource supports the following arguments:
 * `deregistration_delay` - (Optional) Amount time for Elastic Load Balancing to wait before changing the state of a deregistering target from draining to unused. The range is 0-3600 seconds. The default value is 300 seconds.
 * `health_check` - (Optional, Maximum of 1) Health Check configuration block. Detailed below.
 * `lambda_multi_value_headers_enabled` - (Optional) Whether the request and response headers exchanged between the load balancer and the Lambda function include arrays of values or strings. Only applies when `target_type` is `lambda`. Default is `false`.
-* `load_balancing_algorithm_type` - (Optional) Determines how the load balancer selects targets when routing requests. Only applicable for Application Load Balancer Target Groups. The value is `round_robin` or `least_outstanding_requests`. The default is `round_robin`.
+* `load_balancing_algorithm_type` - (Optional) Determines how the load balancer selects targets when routing requests. Only applicable for Application Load Balancer Target Groups. The value is `round_robin`, `least_outstanding_requests`, or `weighted_random`. The default is `round_robin`.
+* `load_balancing_anomaly_mitigation` - (Optional) Determines whether to enable target anomaly mitigation.  Target anomaly mitigation is only supported by the `weighted_random` load balancing algorithm type.  See [doc](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#automatic-target-weights) for more information.  The value is `"on"` or `"off"`. The default is `"off"`.
```

Comment:
It appears the convention might be to name this argument `load_balancing_anomaly_mitigation_enabled` and make it boolean instead?

Reply:
@srhaber thanks for the comment!  I had initially coded this enhancement as such but while testing I found that the AWS API only takes values `on` and `off`.

> load_balancing.algorithm.anomaly_mitigation - Only available when load_balancing.algorithm.type is weighted_random . Indicates whether anomaly mitigation is enabled. The value is on or off . The default is off .

This means we'd have to include code to translate `true` => `on` and `false` => `off`.

I'm not sure why AWS made these `on` and `off` whereas other target group attributes are often `true` and `false`. Perhaps new enumerations will be supported in future?

I'm curious what the terraform maintainers thoughts are.

Reply:
We prefer to match the AWS API as closely as possible, so as-implemented is good.